lib/unbeatable_player.ex

```diff
@@ -1,103 +1,75 @@
-    if new_best_score?(current_mark, computer_mark, new_score, existing_score) do
-      {new_score, next_move}
+  defp check_result(board, move, current_best, depth) do
+    if current_best[:alpha] >= current_best[:beta] do
+      current_best
     else
-      {existing_score, existing_best_move}
-    end
-  end
-
-  defp prune({best_score, best_move}, computer_mark, current_mark, alpha, beta) do
-    new_alpha = update_alpha(alpha, best_score, current_mark, computer_mark)
-    new_beta = update_beta(beta, best_score, current_mark, computer_mark)
-
-    if new_alpha >= new_beta do
-      {:halt, [best_score, best_move]}
-    else
-      {:cont, [best_score, best_move]}
+      board
+      |> Board.place_mark(move)
+      |> negamax(depth - 1, -current_best[:beta], -current_best[:alpha])
+      |> negate_score
+      |> better_score?(current_best)
+      |> update_score(move)
     end
   end
 
-  defp update_alpha(alpha, best_score, current_mark, computer_mark) do
-    if current_mark == computer_mark, do: Enum.max([alpha, best_score]), else: alpha
+  defp update_score({false, _, current_best}, _), do: current_best
+  defp update_score({true, new_result, current_best}, move) do
+    %{
+      :best_score => new_result[:best_score] ,
+      :best_move => move,
+      :alpha => new_result[:best_score],
+      :beta => current_best[:beta],
+    }
   end
 
-  defp update_beta(beta, best_score, current_mark, computer_mark) do
-    if current_mark != computer_mark, do: Enum.min([beta, best_score]), else: beta
+  defp better_score?(new_score, old_score) do
+    {new_score[:best_score] > old_score[:alpha], new_score, old_score}
   end
 
-  def score_for_move(board, computer_mark, depth) do
-    winning_mark = Board.winning_mark(board)
-    cond do
-      winning_mark == computer_mark ->
-        depth
-      Board.draw?(board) ->
-        0
-      winning_mark != computer_mark ->
-        -depth
-    end
+  defp negate_score(result) do
+    %{:best_score => -result[:best_score], :best_move => result[:best_move]}
   end
 
-  defp new_best_score?(current_mark, computer_mark, new_score, best_score) when current_mark == computer_mark do
-    new_score > best_score
+  defp accumulator(alpha, beta) do
+    %{
+      :best_score => -@initial_score,
+      :best_move => @move_placehoder,
+      :alpha => alpha,
+      :beta => beta,
+    }
   end
 
-  defp new_best_score?(current_mark, computer_mark, new_score, best_score) when current_mark != computer_mark do
-    new_score < best_score
+  def score(board, depth) do
+    if Board.winner?(board), do: -(depth + 1), else: 0
   end
 
-  defp switch(current_mark) do
-    if (current_mark == "X"), do: "O", else: "X"
-  end
-
-  defp initialize_score(current_mark, computer_mark) do
-    if (current_mark == computer_mark), do: -1000, else: 1000
-  end
+  defp finish_calculation?(board, depth), do: if Board.game_over?(board) || depth == 0, do: :finish, else: :continue
```

Comment:
OK, I thought the more lines the better! But yes, it's easier to read when it's not flat.